packages/ui/src/components/modals/ActionCallbackResultDropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react'
+import * as CLM from '@conversationlearner/models'
+import * as OF from 'office-ui-fabric-react'
+import * as TC from '../tipComponents'
+import * as ToolTips from '../ToolTips/ToolTips'
+import * as Util from '../../Utils/util'
+import { FM } from '../../react-intl-messages'
+import { injectIntl, InjectedIntlProps } from 'react-intl'
+import CallbackResultViewerModal from './CallbackResultViewerModal'
+import './ActionCallbackResultDropdown.css'
+
+type ReceivedProps = {
+    action: CLM.ApiAction,
+    callback: CLM.Callback | undefined,
+    selectedCallbackResult?: CLM.CallbackResult,
+    onChangeSelectedCallbackResult: (callbackResult: CLM.CallbackResult) => void,
+}
+
+type Props = ReceivedProps & InjectedIntlProps
+
+const Component: React.FC<Props> = (props) => {
+    const [isCallbackResultViewerOpen, setIsCallbackResultViewerOpen] = React.useState(false)
+    const onClickViewCallbackResult = () => {
+        setIsCallbackResultViewerOpen(s => !s)
+    }
+
+    const onClickCancelStubViewer = () => {
+        setIsCallbackResultViewerOpen(false)
+    }
+
+    // Compute list of results from callbacks
+    // If no results defined, show None and Disable preview.
+    // If results defined list stub names and none selected.
+    const noneOptionKey = 'none'
+    const callbackResultOptions: OF.IDropdownOption[] = [
+        {
+            key: noneOptionKey,
+            text: 'None',
+        },
+    ]
+
+    // TODO: Add options from stubs defined on model
+    const callback = props.callback
+    if (callback?.results) {
+        const callbackResultOptionsFromBot = callback.results.map<OF.IDropdownOption>(callbackResult => ({
+            key: callbackResult.name,
+            text: callbackResult.name,
+            data: callbackResult,
+        }))
+
+        callbackResultOptions.push(...callbackResultOptionsFromBot)
+    }
+
+    const [selectedCallbackResultOptionKey, setSelectedCallbackResultOptionKey] = React.useState(props.selectedCallbackResult?.name ?? noneOptionKey)
+    // React.useEffect(() => {
```

Comment:
nit : if this is going to be used in the near future, please add a comment.  Otherwise, please delete.

Reply:
Removed